pkg/syncer/resource_syncer.go

```diff
@@ -175,7 +192,7 @@ func (r *resourceSyncer) processNextWorkItem(key, name, ns string) (bool, error)
 	klog.V(log.DEBUG).Infof("Syncer %q retrieved %sd resource: %#v", r.config.Name, op, resource)
 
 	if r.shouldSync(resource) {
-		resource, requeue := r.transform(resource, op)
+		resource, transformed, requeue := r.transform(resource, op)
```

Comment:
What is this new value for? Currently resource contains the transformed object. With this change is the expectation for resource to be original and transformed to be new?

Reply:
yes - there may not be a transformed resource if there's no callback for it, in which case the successful sync callback will use the original resource.